### PR TITLE
Make DeletePipeline handle missing repos again.

### PR DESCRIPTION
**Pull request by PFedak**

### CODE REVIEW

**Comment by PFedak** [comment]
Seems like some more parts of the delete could/should be in a transaction

File: src/server/pps/server/api_server.go [hunk 1]
```diff
@@ -2535,20 +2540,23 @@ func (a *apiServer) deletePipeline(ctx context.Context, request *pps.DeletePipel
...
-					return pachClient.DeleteRepo(input.Cron.Repo, request.Force)
-				})
-			}
-			return nil
-		})
-	}
-	if err := eg.Wait(); err != nil {
-		return err
+	// only do this if we were able to read the spec file
+	if pipelineInfo.Details != nil {
```
Comment (acohen4) [inline src/server/pps/server/api_server.go:2544]: I think having this makes logical sense. I'm only confused about the pickle the user may be in if we fail to get the pipeline details, but there are in fact cron repos to delete. I guess in that case they would need to manually debug anyway

File: src/server/pps/server/api_server.go [hunk 2]
```diff
@@ -2558,28 +2566,30 @@ func (a *apiServer) deletePipeline(ctx context.Context, request *pps.DeletePipel
...
+	if !missingRepo {
+		if !request.KeepRepo {
+			// delete the pipeline's output repo
+			if _, err := a.env.PfsServer().DeleteRepo(ctx, &pfs.DeleteRepoRequest{
+				Repo:  client.NewRepo(request.Pipeline.Name),
+				Force: request.Force,
+			}); err != nil && !errutil.IsNotFoundError(err) {
+				return errors.Wrap(err, "error deleting pipeline repo")
+			}
+		} else if pipelineInfo.Details != nil {
```
Comment (acohen4) [inline src/server/pps/server/api_server.go:2578]: should we add a final else that throws an error?

**Review by acohen4 — APPROVED** [review]
Left a few comments, but overall looks good!

File: src/server/pps/server/api_server.go (hunk 1 above)
Comment (PFedak) [reply src/server/pps/server/api_server.go:2544]: Yeah, I think if the spec really is gone for whatever reason we're stuck. It is the case that we create cron repos with a certain name pattern, but that doesn't seem safe enough. If I get my cron repos -> system repos change in this goes away, anyway

File: src/server/pps/server/api_server.go (hunk 2 above)
Comment (PFedak) [reply src/server/pps/server/api_server.go:2582]: Good idea

**Comment by msteffen** [comment]
All tests are green, so I'm merging this PR so I can merge https://github.com/pachyderm/pachyderm/pull/6722

**Pull request merged**